packages/core/src/routes/verification/index.ts

```diff
@@ -46,12 +58,96 @@ export default function verificationRoutes<T extends UserRouter>(
         passwordVerification.verify(password)
       );
 
-      await saveVerificationRecord(userId, passwordVerification, queries);
+      await insertVerificationRecord(passwordVerification, queries, userId);
 
       ctx.body = { verificationRecordId: passwordVerification.id };
       ctx.status = 201;
 
       return next();
     }
   );
+
+  router.post(
+    '/verifications/verification-code',
+    koaGuard({
+      body: z.object({
+        identifier: verificationCodeIdentifierGuard,
+      }),
+      response: z.object({ verificationRecordId: z.string() }),
+      status: [201, 501],
+    }),
+    async (ctx, next) => {
+      const { id: userId } = ctx.auth;
+      const { identifier } = ctx.guard.body;
+
+      const user = await queries.users.findUserById(userId);
+
+      const codeVerification = createNewCodeVerificationRecord(
+        libraries,
+        queries,
+        identifier,
+        // TODO(LOG-10148): Add new event
```

Comment:
nit: maybe update the input param to accept an email template instead of an interaction event.

Reply:
Good suggestion, I'll consider it.